File: src/test/java/com/gildedrose/TexttestFixture.java
package com.gildedrose;

import com.gildedrose.utils.EncapsulatedItem;
import com.gildedrose.utils.EncapsulateItemBuilder;

public class TexttestFixture {
    public static void main(String[] args) {
        System.out.println("OMGHAI!");

        final EncapsulatedItem[] items = buildItems();

        GildedRose app = new GildedRose(items);

        int days = 2;
        if (args.length > 0) {
            days = Integer.parseInt(args[0]) + 1;
        }

        for (int i = 0; i < days; i++) {
            System.out.println("-------- day " + i + " --------");
            System.out.println("name, sellIn, quality");
            for (EncapsulatedItem item : items) {
                System.out.println(item);
            }
            System.out.println();
            app.updateQuality();
        }
    }

    private static EncapsulatedItem[] buildItems() {
        return new EncapsulatedItem[] {
                new EncapsulateItemBuilder().named("+5 Dexterity Vest").toSellIn(10).ofQuality(20).build(),
                new EncapsulateItemBuilder().named("Aged Brie").toSellIn(2).ofQuality(0).build(),
                new EncapsulateItemBuilder().named("Elixir of the Mongoose").toSellIn(5).ofQuality(7).build(),
                new EncapsulateItemBuilder().named("Sulfuras, Hand of Ragnaros").toSellIn(0).ofQuality(80).build(),
                new EncapsulateItemBuilder().named("Sulfuras, Hand of Ragnaros").toSellIn(-1).ofQuality(80).build(),
                new EncapsulateItemBuilder().named("Backstage passes to a TAFKAL80ETC concert").toSellIn(15).ofQuality(20).build(),
                new EncapsulateItemBuilder().named("Backstage passes to a TAFKAL80ETC concert").toSellIn(10).ofQuality(49).build(),
                new EncapsulateItemBuilder().named("Backstage passes to a TAFKAL80ETC concert").toSellIn(5).ofQuality(49).build(),
                new EncapsulateItemBuilder().named("Conjured Mana Cake").toSellIn(3).ofQuality(6).build(),};

    }

}

File: scala/src/test/scala/com/gildedrose/TexttestFixture.scala
package com.gildedrose

object TexttestFixture {
  def main(args: Array[String]): Unit = {
    System.out.println("OMGHAI!")

    val items = Array[Item](
        Item("+5 Dexterity Vest", 10, 20),
        Item("Aged Brie", 2, 0),
        Item("Elixir of the Mongoose", 5, 7),
        Item("Sulfuras, Hand of Ragnaros", 0, 80),
        Item("Sulfuras, Hand of Ragnaros", -1, 80),
        Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
        Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
        Item("Backstage passes to a TAFKAL80ETC concert", 5, 49), // this conjured item does not work properly yet
        Item("Conjured Mana Cake", 3, 6)
    )

    val app  = new GildedRose(items)
    var days = 2

    if (args.nonEmpty) days = args(0).toInt + 1

    for (i <- 0 until days) {
      System.out.println("-------- day " + i + " --------")
      System.out.println("name, sellIn, quality")
      for (item <- items) {
        System.out.println(s"${item.name}, ${item.sellIn}, ${item.quality}")
      }
      System.out.println()
      app.updateQuality()
    }
  }
}
